File: 0201_0300/0267/0267.py
from math import gcd
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n, x, y = (int(x) for x in f.readline().split())
    x, y = (y, x) if x > y else (x, y)
    n -= 1
    p = x * y // gcd(x, y)
    k = p // y  + p // x
    t = x + n // k * p
    n %= k
    p = 0
    while n:
        if n * x - p <= y:
            t += n * x - p
            break
        else:
            t += y
            n -= 1 + (y + p) // x
            p = (y + p) % x
    q.write(str(t))

File: 0201_0300/0267/0267.c
#include <stdio.h>

int gcd(int m, int n)
{
	while (n && m)
		if (n > m)
			n %= m;
		else
			m %= n;
	return n + m;
}

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int n, x, y, t, k, p;

	fscanf(f, "%d%d%d", &n, &x, &y);
	if (x > y)
		x ^= y, y ^= x, x ^= y;
	--n, t = x;
	p = x * y / gcd(x, y);
	k = p / y + p / x;
	t += n / k * p, n %= k;
	p = 0;
	while (n)
		if (n * x - p <= y)
			t += n * x - p, n = 0;
		else
			t += y, n -= 1 + (y + p) / x, p = (y + p) % x;
	fprintf(q, "%d", t);
	return 0;
}
